### PR TITLE
Google cloud: make our recommendations more clear

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: src/server/google-cloud.md [hunk 1]
```diff
@@ -15,7 +15,11 @@ For information about other Google APIs (including Firebase)
 that you might want to use from Dart code,
 see the [Google APIs page](/guides/google-apis).
 
-## Cloud Run
+## Recommended solutions
+
+To run Dart in the Cloud, we recommand using serverless computing solutions.
```
Comment (kwalrath) [inline src/server/google-cloud.md:20]:
```suggestion
To run Dart in the Cloud, we recommend using serverless computing solutions.
```

**Review by kwalrath — APPROVED** [review]
After fixing a typo, LGTM if it LGT @mit-mit

File: src/server/google-cloud.md (hunk 1 above)
Comment (kevmoo) [reply src/server/google-cloud.md]: I guess...if you care about..."spelling" 🤷  🙃

File: src/server/google-cloud.md [hunk 2]
```diff
@@ -47,14 +56,14 @@ combined with Dart's Docker images.
...
 
 To run Dart on clusters of Compute Engine instances,
 use Google Kubernetes Engine (GKE).
 
 For more information, see the [GKE overview][].
 
-## App Engine
+### App Engine
 
 [App Engine][] support for Dart is incomplete,
```
Comment (mit-mit) [inline src/server/google-cloud.md:68]: Do we want to be even more specific here? E.g.:

"... is incomplete, and requires using AppEngine's flex environment, which will not automatically [scale to zero](https://cloud.google.com/run/docs/about-instance-autoscaling), so we..."

Comment (kevmoo) [reply src/server/google-cloud.md]: PTAL

File: src/server/google-cloud.md [hunk 3]
```diff
@@ -47,22 +56,25 @@ combined with Dart's Docker images.
...
 For more information, see the [GKE overview][].
 
-## App Engine
+### App Engine
 
-[App Engine][] support for Dart is incomplete,
-so we recommend that you **use Cloud Run** instead of App Engine for new
+[App Engine][] support for Dart is incomplete and requires the
+[App Engine Flexible Environment][] which does not
+["scale to zero"][scale to zero], so we recommend **Cloud Run** for new
```
Comment (kwalrath) [inline src/server/google-cloud.md:70]:
```suggestion
[autoscale to zero instances][scale to zero], so we recommend **Cloud Run** for new
```

File: src/server/google-cloud.md [hunk 4]
```diff
@@ -47,22 +56,25 @@ combined with Dart's Docker images.
...
 
 For more information, see the [GKE overview][].
 
-## App Engine
+### App Engine
 
-[App Engine][] support for Dart is incomplete,
-so we recommend that you **use Cloud Run** instead of App Engine for new
+[App Engine][] support for Dart is incomplete and requires the
+[App Engine Flexible Environment][] which does not
```
Comment (kwalrath) [inline src/server/google-cloud.md:69]:
```suggestion
[App Engine flexible environment][], which does not
```

File: src/server/google-cloud.md [hunk 5]
```diff
@@ -47,22 +56,25 @@ combined with Dart's Docker images.
...
+["scale to zero"][scale to zero], so we recommend **Cloud Run** for new
 server-side Dart code.
 If you _want_ to use App Engine, consider using the [`appengine` package][].
 
 
 [App Engine]: https://cloud.google.com/appengine
+[App Engine Flexible Environment]: https://cloud.google.com/appengine/docs/flexible
```
Comment (kwalrath) [inline src/server/google-cloud.md:76]:
```suggestion
[App Engine flexible environment]: https://cloud.google.com/appengine/docs/flexible
```

**Review by mit-mit — APPROVED** [review]

**Pull request merged**